openshift-kube-apiserver/admission/customresourcevalidation/authentication/validate_authentication.go

```diff
@@ -46,21 +58,34 @@ func toAuthenticationV1(uncastObj runtime.Object) (*configv1.Authentication, fie
 	return obj, nil
 }
 
-type authenticationV1 struct{}
+type celStore struct {
+	compilingGroup *singleflight.Group
+	compiledStore  *lru.Cache
+	compiler       authenticationcel.Compiler
+}
+
+type authenticationV1 struct {
+	cel *celStore
+}
 
-func (authenticationV1) ValidateCreate(_ context.Context, uncastObj runtime.Object) field.ErrorList {
+func (a authenticationV1) ValidateCreate(ctx context.Context, uncastObj runtime.Object) field.ErrorList {
 	obj, errs := toAuthenticationV1(uncastObj)
 	if len(errs) > 0 {
 		return errs
 	}
 
 	errs = append(errs, validation.ValidateObjectMeta(&obj.ObjectMeta, false, crvalidation.RequireNameCluster, field.NewPath("metadata"))...)
-	errs = append(errs, validateAuthenticationSpecCreate(obj.Spec)...)
+	err, warns := validateAuthenticationSpecCreate(ctx, obj.Spec, a.cel)
+	errs = append(errs, err...)
+	for _, warn := range warns {
+		warning.AddWarning(ctx, "", warn)
+		klog.Warning(warn)
+	}
```

Comment:
In the latest revision the context is now wired all the way down to where the warning is generated... can you get rid of all the plumbing and handle the warning closer to the source?